plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/ResourceMojo.java

```diff
@@ -417,30 +419,36 @@ public boolean accept(File pathname) {
     }
 
     private KubernetesListBuilder generateAppResources(List<ImageConfiguration> images, EnricherManager enricherManager) throws IOException, MojoExecutionException {
-        File[] resourceFiles = KubernetesResourceUtil.listResourceFragments(resourceDir);
-        KubernetesListBuilder builder;
+        try {
+            File[] resourceFiles = KubernetesResourceUtil.listResourceFragments(resourceDir);
+            KubernetesListBuilder builder;
 
-        // Add resource files found in the fabric8 directory
-        if (resourceFiles != null && resourceFiles.length > 0) {
-            log.info("Using resource templates from %s", resourceDir);
-            builder = readResourceFragments(resourceFiles);
-        } else {
-            builder = new KubernetesListBuilder();
-        }
+            // Add resource files found in the fabric8 directory
+            if (resourceFiles != null && resourceFiles.length > 0) {
+                log.info("Using resource templates from %s", resourceDir);
+                builder = readResourceFragments(resourceFiles);
+            } else {
+                builder = new KubernetesListBuilder();
+            }
 
-        // Add locally configured objects
-        if (resources != null) {
-            // TODO: Allow also support resources to be specified via XML
-            addConfiguredResources(builder, images);
-        }
+            // Add locally configured objects
+            if (resources != null) {
+                // TODO: Allow also support resources to be specified via XML
+                addConfiguredResources(builder, images);
+            }
 
-        // Create default resources for app resources only
-        enricherManager.createDefaultResources(builder);
+            // Create default resources for app resources only
+            enricherManager.createDefaultResources(builder);
 
-        // Enrich descriptors
-        enricherManager.enrich(builder);
+            // Enrich descriptors
+            enricherManager.enrich(builder);
 
-        return builder;
+            return builder;
+        } catch (ConstraintViolationException e) {
+            String message = ValidationUtil.createValidationMessage(e.getConstraintViolations());
+            getLog().error("ConstraintViolationException: " + message, e);
```

Comment:
whether or not you supply a message to `new ConstraintViolationException()`  the violation objects are available via `getConstraintViolations()` so that any downstream consumer can format the violations however they want. 
So thats not an argument for turning the ConstraintViolationException exception into a meaningless exception with no message at all ;)
A library could even catch the exception, change the message format and rethrow if it really wished to do so.